test/e2e/storage/vsphere_volume_disksize.go

```diff
@@ -0,0 +1,99 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	DiskSizeSCName = "disksizesc"
+)
+
+/*
+	Test to verify disk size specified in PVC is being honored while volume creation.
+
+	Steps
+	1. Create StorageClass.
+	2. Create PVC with invalid disk size which uses the StorageClass created in step 1.
+	3. Expect the PVC to fail.
+	4. Verify the error returned on PVC failure is the correct.
+*/
+
+var _ = SIGDescribe("Verify vsphere persistent volume creation with disksize on a datastore", func() {
+	f := framework.NewDefaultFramework("volume-disksize")
+	var (
+		client       clientset.Interface
+		namespace    string
+		scParameters map[string]string
+		datastore    string
+	)
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("vsphere")
+		client = f.ClientSet
+		namespace = f.Namespace.Name
+		scParameters = make(map[string]string)
+		datastore = os.Getenv("VSPHERE_DATASTORE")
+		Expect(datastore).NotTo(BeEmpty())
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+	})
+
+	It("verify dynamically provisioned pv using storageclass with an invalid disk size fails", func() {
+		By("Invoking Test for invalid disk size")
+		scParameters[Datastore] = datastore
+		scParameters[DiskFormat] = ThinDisk
+		diskSize := "1"
+		err := invokeInvalidDiskSizeTestNeg(client, namespace, scParameters, diskSize)
+		Expect(err).To(HaveOccurred())
+		errorMsg := `Failed to provision volume with StorageClass \"` + DiskSizeSCName + `\": A specified parameter was not correct`
+		if !strings.Contains(err.Error(), errorMsg) {
+			Expect(err).NotTo(HaveOccurred(), errorMsg)
+		}
+	})
+})
+
+func invokeInvalidDiskSizeTestNeg(client clientset.Interface, namespace string, scParameters map[string]string, diskSize string) error {
+	By("Creating Storage Class With invalid disk size")
+	storageclass, err := client.StorageV1().StorageClasses().Create(getVSphereStorageClassSpec(DiskSizeSCName, scParameters))
+	Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to create storage class with err: %v", err))
+	defer client.StorageV1().StorageClasses().Delete(storageclass.Name, nil)
+
+	By("Creating PVC using the Storage Class")
+	pvclaim, err := framework.CreatePVC(client, namespace, getVSphereClaimSpecWithStorageClassAnnotation(namespace, diskSize, storageclass))
+	Expect(err).NotTo(HaveOccurred())
+	defer framework.DeletePersistentVolumeClaim(client, pvclaim.Name, namespace)
+
+	By("Waiting for claim to be in bound phase")
```

Comment:
By("Waiting for claim to be in bound phase") => By("Expect claim to fail provisioning volume")

Reply:
Fixed!